### PR TITLE
feat(#535): Add basic text search for booking requests

### LINKED ISSUE #535: Add filter to my booking page
Add filter to my booking page

### PR DESCRIPTION
**Pull request by EvanDime**
##### **_IMPORTANT: Always create an issue before a PR and provide enough information so that others can review_**

Adding a filter for the "My Bookings" Page 

##### _What existing problem does the pull request solve?_

when you type in the text search it will filter the list of bookings listed. 

##### _Demonstrate the code is solid. </br> Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI._

Back end test to test the api controller. 
jest test that checks every aspect of the added. computed data, axios, etc. 

##### _List all issues this PR resolves_
- closes #535 

##### _Complete these items before requesting reviews_
- [x] Static Analysis Rules Validated
- [x] All commits follow naming conventions

### Sub-tasks Checklist
- [x] UI Implementation
- [x] UI Test
- [x] Backend Implementation
- [x] Backend Tests
- [x] Integration Tests
- [x] System Tests
- [x] Documentation

### CODE REVIEW

**Review by joeyabou — DISMISSED** [review]
Good job adding this the stakeholders really seem to want this when I meet with them. Only thing left is to fill out the description of the PR and seems good to go !

**Comment by EvanDime** [comment]
Please do not listen to the sonar cloud lies...i have 100% code coverage and i'll fight anyone who tries to disagree with me

File: app/Http/Controllers/BookingRequestController.php
Comment (alexstojda) [inline app/Http/Controllers/BookingRequestController.php:362]: `myFilter` is a bit ambiguous for a function name. Could you change it to something more descriptive, so as to better differentiate it from the `filter` function.

File: tests/Javascript/Components/Tables/UsersBookingRequestTable.spec.js
Comment (alexstojda) [inline tests/Javascript/Components/Tables/UsersBookingRequestTable.spec.js:1148]: Could you move these large `consts` either outside of the tests or in a separate file? It will make the tests easier to read

**Review by alexstojda — CHANGES_REQUESTED** [review]
couple small things :)

File: app/Http/Controllers/BookingRequestController.php
Comment (EvanDime) [reply app/Http/Controllers/BookingRequestController.php:362]: i figured using myFilter would be a good name for the filter that is used on a user looking at their own bookings but sure i'll change it

Comment (EvanDime) [reply app/Http/Controllers/BookingRequestController.php]: Name updated

File: tests/Javascript/Components/Tables/UsersBookingRequestTable.spec.js
Comment (EvanDime) [reply tests/Javascript/Components/Tables/UsersBookingRequestTable.spec.js]: Moved

File: app/Models/BookingRequest.php
Comment (a-drew) [inline app/Models/BookingRequest.php:100]: maybe this should be split into 2 different scopes?

Comment (EvanDime) [reply app/Models/BookingRequest.php:100]: i disagree since a booking that is in review is still a pending review

File: resources/js/Components/Tables/BookingsTable.vue
Comment (a-drew) [inline resources/js/Components/Tables/BookingsTable.vue:69]: might be cleaner to do the v-if else-if logic on the button and not repeat the <a :href> tag 3 times

**Review by a-drew — APPROVED** [review]
couple of small refactors you can make to improve the code but looks good. I'll let you decide if you want to implement the changes or not.

File: resources/js/Components/Tables/BookingsTable.vue
Comment (EvanDime) [reply resources/js/Components/Tables/BookingsTable.vue:69]: what is the difference? Each button has it's own class anyway. it will still be repeated 3-4 times. At least this way if we need to change anything about each link, we are able to.

File: app/Models/BookingRequest.php
Comment (a-drew) [reply app/Models/BookingRequest.php:100]: sounds fair

File: resources/js/Components/Tables/BookingsTable.vue
Comment (EvanDime) [reply resources/js/Components/Tables/BookingsTable.vue:69]: like if in the future, denied bookings can maybe redirect back to an edit page so that the user can try again.

Comment (a-drew) [reply resources/js/Components/Tables/BookingsTable.vue:69]: Right now you're repeating the href navigation path 3 times as well as the buttons. If you refactor you'll only be repeating the thing that is actually different aka "button with different color and text"

Comment (EvanDime) [reply resources/js/Components/Tables/BookingsTable.vue]: applied changes

**Review by a-drew — APPROVED** [review]
@EvanDime can you update from master before I approve? I think it'll dismiss my review again

**Comment by EvanDime** [comment]
> @EvanDime can you update from master before I approve? I think it'll dismiss my review again

It won't since i plan on using the update branch button from github. if i manually did a rebase and force push form my IDE then it would ruin it.

**Review by alexstojda — APPROVED** [review]

**Pull request merged**
